osu.Game.Rulesets.Osu/Difficulty/Skills/OsuStrainSkill.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using osu.Game.Rulesets.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Difficulty.Skills;
+using osu.Game.Rulesets.Mods;
+using System.Linq;
+
+namespace osu.Game.Rulesets.Osu.Difficulty.Skills
+{
+    public abstract class OsuStrainSkill : StrainSkill
+    {
+        protected virtual int ReducedSectionCount => 10;
+        protected virtual double ReducedStrainBaseline => 0.7;
+        protected virtual double DifficultyMultiplier => 1.08;
+
+        public OsuStrainSkill(Mod[] mods) : base(mods)
+        {
+        }
+
+        public double OsuDifficultyValue()
```

Comment:
As stated above already, this most definitely cannot persist in this state. `DifficultyValue()` should be overridden in order to return this value.

I'm not sure I follow as to what was meant by [this comment](https://github.com/ppy/osu/pull/13483#discussion_r651144072). The existing strains should not be modified when this is called, since in the future this method may be called multiple times during the beatmap's progress and not necessarily at the end of it, and so this method should be pure (i.e. have no side effects/modify any state).

If a multiplier is to applied to each strain somehow, then I'd recommend storing the multipliers independently and only applying them at the end when returning the final difficulty value. Either that, or allowing overriding `GetCurrentStrainPeaks()` itself.

Reply:
Changed this to override `DifficultyValue()`.

I kept the method pure as you said.